code/script.js

```diff
@@ -1,67 +1,273 @@
+  const nameForm = document.getElementById('name-form')
+  const inputWrapper = document.getElementById('input-wrapper')
+  
+  // If you need any global variables that you can use across different functions, declare them here:
 
+  let counter = 1
 
-// If you need any global variables that you can use across different functions, declare them here:
+  // Declare your functions after this comment
 
+  const botReplay = (message) => {
+    showMessage(message, 'bot')
+  }
 
-// Declare your functions after this comment
+  const userReplay = (message) => {
+    showMessage(message, 'user')
+  }
 
-// This function will add a chat bubble in the correct place based on who the sender is
-const showMessage = (message, sender) => {
-  // the if statement checks if the sender is 'user' and if that's the case it inserts an html senction inside the chat with the posted message
-  if (sender === 'user') {
-    chat.innerHTML += `
-      <section class="user-msg">
-        <div class="bubble user-bubble">
-          <p>${message}</p>
-        </div>
-        <img src="assets/user.png" alt="User" />  
-      </section>
+  const resetCounter = () => {
+    userReplay('Reset')
+    setTimeout (() => botReplay('What can I do for you then?'), 750)
+    counter = 2
+    const message = 'Reset'
+    nextMessage()
+    //nextMessage(message)
+  };
+  
+  const backCounter = () => {
+    //userReplay('Back')
+    //counter = 4
+   counter = 3
+    const message = 'Back'
+    //nextMessage(message)
+    setTimeout(() => nextMessage('Back'), 750)
+  };
+  
+  const nextMessage = (message) => {
+    console.log('counter', counter)
+    if (counter == 2) {
+      console.log("test1")
+     // userReplay(message)
+      nameInput.value = ''
+      setTimeout (showOptions(), 750)
+    
+    } else if (counter == 3 ) {
+      console.log("test2")
+      userReplay(message)
+      nameInput.value = ''
+      setTimeout (() => showPlans('Plan'), 750)
+      
+    } else if (counter == 4 ) {
+      console.log('test4')
+      userReplay(message)
+      nameInput.value= ''
+      setTimeout (() => showPlansABC(message), 750)
+      
+    }
+  }
+
+  const handleNameInput = (event) => {      // Here we creat a function to avoid the submit when we click or press the button sumbit.
+    event.preventDefault()                  // .preventDefault() avoits the refresh the page.
+    const name = nameInput.value            // we create a new variable with the typed message in the input bar to use for the chatbot.
+    userReplay(name)               // here calls the function showMessage() to see what was typed before.
+    nameInput.value = ''                    
+    //counter += 1
+    setTimeout(() => showMessage(`Welcome ${name}. Want to talk about you Plan, or do you have some problem?`, 'bot'), 750)
+    setTimeout(() => nextMessage(showOptions()),750)
+    console.log('counter', counter)
+  };
+
```

Comment:
the counter makes looping easier when it comes to naming, however, understanding this is the loop that actually takes place at the end, naming - rest, back, it might be clearer for readers to give an explanatory comment at line 23.